static/js/containers/ChannelPage_test.js

```diff
@@ -91,168 +159,248 @@ describe("ChannelPage", () => {
-      CLEAR_CHANNEL_ERROR,
-      CLEAR_POST_ERROR
-    ])
-    assert.isUndefined(helper.store.getState().channels.error)
+    assert.isUndefined(store.getState().channels.error)
   })
 
   it("should show a 404 if the channel is not found", async () => {
-    helper.getChannelStub.returns(Promise.reject({ errorStatusCode: 404 }))
-
-    const [wrapper] = await renderComponent(channelURL(currentChannel.name), [
-      actions.channels.get.requestType,
-      actions.channels.get.failureType,
-      actions.subscribedChannels.get.requestType,
-      actions.subscribedChannels.get.successType,
-      SET_CHANNEL_DATA
-    ])
-    wrapper.update()
-    assert(wrapper.find(NotFound).exists())
+    const { inner } = await render({
+      channels: {
+        error: {
+          errorStatusCode: 404
+        }
+      }
+    })
+    assert(inner.find(NotFound).exists())
   })
 
   it("should show an 'unauthorized' if the user is not authorized", async () => {
-    helper.getChannelStub.returns(Promise.reject({ errorStatusCode: 403 }))
-
-    const [wrapper] = await renderComponent(channelURL(currentChannel.name), [
-      actions.channels.get.requestType,
-      actions.channels.get.failureType,
-      actions.subscribedChannels.get.requestType,
-      actions.subscribedChannels.get.successType,
-      SET_CHANNEL_DATA
-    ])
-    wrapper.update()
-    assert(wrapper.find(NotAuthorized).exists())
+    const { inner } = await render({
+      channels: {
+        error: {
+          errorStatusCode: 403
+        }
+      }
+    })
+    assert(inner.find(NotAuthorized).exists())
   })
 
   it("should show a normal error for other error codes", async () => {
-    helper.getChannelStub.returns(Promise.reject({ errorStatusCode: 500 }))
-
-    const [wrapper] = await renderComponent(channelURL(currentChannel.name), [
-      actions.channels.get.requestType,
-      actions.channels.get.failureType,
-      actions.subscribedChannels.get.requestType,
-      actions.subscribedChannels.get.successType,
-      SET_CHANNEL_DATA
-    ])
-    assert.isFalse(wrapper.find(NotFound).exists())
-    assert.include(
-      wrapper
-        .find(".main-content")
-        .at(0)
-        .text(),
-      "Error loading page"
-    )
+    const { inner } = await render({
+      channels: {
+        error: {
+          errorStatusCode: 500
+        }
+      }
+    })
+    assert.isFalse(inner.find(NotFound).exists())
+    assert.include(inner.find(".errored").text(), "Error loading page")
+  })
+
+  it(`sets postsForChannelProcessing`, async () => {
+    const { wrapper } = await render({
+      postsForChannel: {
+        processing: true,
+        loaded:     false
+      }
+    })
+    assert.isTrue(wrapper.props().postsForChannelProcessing)
+  })
+
+  describe("pagination", () => {
+    [true, false].forEach(hasPagination => {
+      it(`has both PostList and InfiniteScroll if there is ${
+        hasPagination ? "" : "no "
+      }pagination`, async () => {
```

Comment:
This can be parametrized to avoid the boolean flags.

```javascript
[
  [{sort: "new"}, 1, "not null"]
  [null, 0, "null"],
].forEach([paginationParam, expectedInfScroll, paramDesc] => {
      it(`has both PostList and InfiniteScroll if pagination param is ${paramDesc}`, async () => {
        //...
        assert.equal(inner.find("InfiniteScroll").length, expectedInfScroll)
      })
}
```